proto/api_test.go

```diff
@@ -150,10 +150,10 @@ func TestCombinable(t *testing.T) {
 
 func TestSetGoErrorCopy(t *testing.T) {
 	rh := ResponseHeader{}
-	err := &Error{Message: "test123"}
-	rh.SetGoError(err)
-	err.Message = "321tset"
-	if rh.Error.Message != "test123" {
+	oErr := &Error{Message: "test123"}
+	rh.Error = oErr
+	rh.SetGoError(&testError{})
```

Comment:
OK, I read this again, and I still disagree. This test was indeed testing that SetGoError was copying the argument - look at it again:
```go
err := &Error{Message: "test123"} // make a proto.Error
rh.SetGoError(err) // set it
err.Message = "321tset" // mutate the original
if rh.Error.Message != "test123" { // confirm the thing that was set was not modified
```
Pretty sure the new test is incorrect